### DISCUSSION TITLE
Preserving backward compatibility after migrating to JustJ build pages.

### DISCUSSION THREAD

**Discussion by ewillink** [eclipse-justj/justj.tools#8] [Q&A]
Background: When EMF upgraded to JustJ technology, Tycho target platform references such as

```
    <repository location="http://download.eclipse.org/modeling/emf/emf/updates/nightly/core"/>
    <repository location="http://download.eclipse.org/modeling/emf/emf/updates/2.14milestones/core"/>
```
had to be changed to

```
    <repository location="https://download.eclipse.org/modeling/emf/emf/builds/nightly"/>
    <repository location="https://download.eclipse.org/modeling/emf/emf/builds/milestone"/>
```

causing the usual releng game of hunt-the-missing-p2-repo.

For my migrations I have the simpler migration `.../updates/releases/...` to `.../builds/release/...`

"builds"/"updates" is a free choice, but "builds" seems better since the new site unifies both updates and drops.

"release"/"releases" is not currently parameterisable, but a new parameter to allow a willful change of spelling wrt EMF seems unwise.

So my query/discussion:

Is it a good idea to rework the pre-migration updates and drops with links to the builds?

**Comment by merks** [comment]
EMF was charged long ago I’m not planning to rework anything for EMF.

**Reply by merks** [reply]
Note too that I keep moving your issues and discussions too the tools repo where the code lives.

**Reply by ewillink** [reply]
Ah! That's why I receive repeats.

I'm not suggesting that EMF changes now; relengs have long since played the game. The refactoring to remove "core" was less trivial. EMF may however want to join in an effort to ensure that the PHP behind e.g. http://www.eclipse.org/modeling/emf/downloads/?project=emf is bypassed allowing users to see the new builds page.

I'm questioning the advisability of links for 'new' migrations.

**Reply by merks** [reply]
I'm having trouble keeping up with "migrate off of the wiki" and "don't use PHP anymore".  Clearly the entire modeling pages is going to need a complete overhaul and I don't think I have the capacity for that, let alone to patch up whats already currently outdated and/or broken.

**Reply by ewillink** [reply]
wrt PHP. Yes. It's a pain, but hierarchical 'downloads.php' has been a bit broken for 10 years and just gets worse and worse with each EF innovation. Quite a few projects have jumped ship with ad hoc downloads pages. I waited for a new 'standard'. 'Fortunately' mending the PHP is now a clear waste of time and fortunately, as I previously observed, the JustJ technology is ten steps forward and one back. So let JustJ provide everything new, then we just need 'redirects' so that readers of old documentation with old URLs migrate to the new. Potentially for http://www.eclipse.org/modeling/emf/downloads/?project=emf we need perhaps a prefix test in perhaps modeling/downloads.php for known/declared JustJ'd projects so that they redirect via the download button in the PMI rather than falling through to legacy functionality. Or maybe modeling/emf/downloads.php is a trivial 100% redirect. Unfortunately the increased security means that I cannot browse to see what we are actually doing (writable/www.eclipse.org is not readable by genies).

wrt migrate off the Wiki, I have yet to see how a replacment system will allow me (as an ECA) to edit e.g. the Tycho FAQ; that is the heart of Open Source. Waiting for busy Tycho developers to moderate a PR is not acceptable for a 'wiki'.

**Reply by ewillink** [reply]
wrt complete overhaul, AFAIAC there are two areas. downloads.php still serves important content for some projects; JustJ can save us. The rest of the php is an embarrassing perpetuation of stale containerised ideas. As a first pass a cut and paste into a few conventional modeling web pages could endeavour to focus on relevant domains such as Ecore modeling, UML modeling, T2M, M2T, M2M , ... with very brief summaries before redirecting to projects. (Maybe we can even persuade Wayne to give us back M2M since IoT doesn't want it and has welshed on its promise to provide a top level MMT reference.)

**Reply by ewillink** [reply]
https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4394 raised to progress downloads.php redirects.
